osu.Game/Modes/ScoreProcesssor.cs

```diff
@@ -46,6 +62,11 @@ public void AddJudgement(JudgementInfo judgement)
             UpdateCalculations(judgement);
 
             judgement.ComboAtHit = (ulong)Combo.Value;
+            if (Health.Value == Health.MinValue && !HasFailed)
+            {
+                HasFailed = true;
+                Failed?.Invoke();

```

Comment:
Since `Failed` event should be invoked whenever `HasFailed` becomes `true`, I think it is good idea to create `HasFailed` property and move this `Invoke()` call to it's setter.